AwsCryptographicMaterialProviders/dafny/AwsCryptographicMaterialProviders/test/TestErrorMessages.dfy

```diff
@@ -0,0 +1,64 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+include "../src/Index.dfy"
+include "../src/ErrorMessages.dfy"
+
+module TestErrorMessages {
+  import Types = AwsCryptographyMaterialProvidersTypes
+  import ErrorMessages
+  import UTF8
+  import AS = AlgorithmSuites
+
+  const TEST_DBE_ALG_SUITE_ID := Types.AlgorithmSuiteId.DBE(Types.ALG_AES_256_GCM_HKDF_SHA512_COMMIT_KEY_SYMSIG_HMAC_SHA384)
+
+  method {:test} TestIncorrectRawDataKeys()
+  {
+    var datakey := "0";
+    var keyringName := "RSAKeyring";
+    var keyProviderId := "TestProvider";
+    var actualErrorMessage := ErrorMessages.IncorrectRawDataKeys(datakey, keyringName, keyProviderId);
+    var ExpectErrorMessage := "EncryptedDataKey "
```

Comment:
What I'd like to see is
ExpectErrorMessage :=
"0 did not match RSAKeyring. Expected: keyProviderId: TestProvider.\n"
because
a) It's really hard for me to look at the test and deduce that string
b) I think the string you're producing is actually
"0 did not matchRSAKeyring. Expected: keyProviderId: TestProvider.\n"